tests/unit/command/ls/test_ls_colors.py

```diff
@@ -0,0 +1,58 @@
+from dvc.command.ls.ls_colors import LsColors
+
+
+def colorize(ls_colors):
+    def _colorize(f, spec=""):
+        fs_path = {
+            "path": f,
+            "isexec": "e" in spec,
+            "isdir": "d" in spec,
+            "isout": "o" in spec,
+        }
+        return ls_colors.format(fs_path)
+
+    return _colorize
+
+
+def test_ls_colors_out_file():
+    ls_colors = LsColors(LsColors.default)
+    assert colorize(ls_colors)("file", "o") == "file"
```

Comment:
May use the same notation in `Repo.ls()`, also may use "/" as sep and then just str replace it, drop all those parentheses. Up to you.

This will also help you test isdir and isexec parts seamlessly.

Reply:
> May use the same notation in Repo.ls(), also may use "/" as sep and then just str replace it, drop all those parentheses. Up to you.

I didn't get your point, could you please extend it? Or maybe an example of your idea?

Reply:
You can use:

```python
    match_files(
        files,
        [     
            ("some-dir/statistics/data.csv", "o"),
            ("some-dir/subdir", "d"),
            ("some-dir/subdir", "d"),
            ("derived", "do"),
        ]
    )
```

I.e. the same DSL for testing `Repo.ls()`, should make it more readable.

Reply:
Yes, it looks really nice, but let me keep as it is at cause:
1. Only one method in `tests/func/test_ls` really tests coloring, all others don't output colors (`isatty = False`)
2. The improvement requires a lot of efforts and doesn't bring real value at this point
2. I think it has sense when we add more types (`ln`, `broken ln`) and decide to cover coloring in `func/test_ls` also

Reply:
Up to you, this has nothing to do with coloring BTW.